src/wp-includes/formatting.php

```diff
@@ -5858,36 +5858,32 @@ function wp_spaces_regexp() {
 }
 
 /**
- * Prints the important emoji-related styles.
+ * Enqueue the important emoji-related styles.
  *
- * @since 4.2.0
+ * @since 6.4.0
  */
-function print_emoji_styles() {
-	static $printed = false;
-
-	if ( $printed ) {
+function wp_enqueue_emoji_styles() {
+	$action = is_admin() ? 'admin_print_styles' : 'wp_print_styles';
+	if ( ! has_action( $action, 'print_emoji_styles' ) ) {
 		return;
 	}
-
-	$printed = true;
-
-	$type_attr = current_theme_supports( 'html5', 'style' ) ? '' : ' type="text/css"';
-	?>
-<style<?php echo $type_attr; ?>>
-img.wp-smiley,
-img.emoji {
-	display: inline !important;
-	border: none !important;
-	box-shadow: none !important;
-	height: 1em !important;
-	width: 1em !important;
-	margin: 0 0.07em !important;
-	vertical-align: -0.1em !important;
-	background: none !important;
-	padding: 0 !important;
-}
-</style>
-	<?php
+	remove_action( $action, 'print_emoji_styles' );
+	$emoji_styles = /* language=CSS */ '
```

Comment:
Adding "paragraph break" to isolate from above back-compat code.

```suggestion

	$emoji_styles = /* language=CSS */ '
```